go/merge/three_way.go

```diff
@@ -11,6 +11,8 @@ import (
 	"github.com/attic-labs/noms/go/types"
 )
 
+type ResolveFunc func(aChange, bChange types.ValueChanged, a, b types.Value, path types.Path) (change types.ValueChanged, merged types.Value, ok bool)
```

Comment:
// ResolveFunc is the type of a function which takes two conflicting changes and tries to resolve them. The two changes occured at `path`, one resulting in the Value `a` and the other in the Value `b`. If a resolution cannot be devised, `ok` should be false upon return and the other return values are undefined. If the conflict can be resolved, the function should return the appropriate types.ValueChanged to apply, the new value to be used when applying the change, and `true`.

(I will put this in the source too, obviously, but writing it here to hopefully explain)